### ISSUE TITLE
Implement more complex HTTP POST... requests with custom body

### ISSUE THREAD

**Issue by SciLor** [phpservermon/phpservermon#638]
I have a JSON Webservice where I am expecting a specific result when I send a specific answer.

Positive case:
HTTP 400
{ "code": 123 }

Negative case:
HTTP 400
{ "code": 75 }

A workaround could be to allow a shellscript to make the check.

**Comment by TimZ99** [comment]
#628 lets you mark HTTP 400 as online.
For the JSON part; why not use the pattern check?

**Comment by SciLor** [comment]
I have to send a specific POST request to check if the service is online.
#628 would be also needed.

As I said as workaround I created a shellscript that is called by a php script that is checked by psm.

**Comment by TimZ99** [comment]
#631 is also needed for this. Only an input field is needed and it's done.

**Comment by SciLor** [comment]
Thank you for your quick answers. You may mark this one as duplicate of the others.

**Comment by TimZ99** [comment]
It isn't a duplicate. The other issues don't include a post field.

Could you review #628 and #631? They need to be merged first.

**Comment by SciLor** [comment]
What do you expect by review? Never did that before. They both look legit for me.

**Comment by TimZ99** [comment]
Testing if the pr works as intended.

**Comment by TimZ99** [comment]
If you use git, it's as easy as git checkout 'the branch' and opening your web browser.

**Comment by SciLor** [comment]
Then just copy over the files to an (backuped) installation and check if its fine? I will try that on the weekend latest.

**Comment by TimZ99** [comment]
You know what, I just merge them :sparkles:. That way I can work on this issue now :smile:.

**Comment by TimZ99** [comment]
It's done btw :P

**Comment by SciLor** [comment]
Can I just checkout the git and install it to my server?

**Comment by TimZ99** [comment]
How do you have PSM installed? Did you clone using git or download the zip file?

**Comment by SciLor** [comment]
I installed it via the zip file.

**Comment by TimZ99** [comment]
- Download https://github.com/phpservermon/phpservermon/archive/develop.zip
- Remove all files from your PSM folder except config.php.
- Move files from the download to you PSM folder.
- Open PSM.

Please be aware that this version is not yet released and should be considered a beta.

**Comment by SciLor** [comment]
I used the release zip for the installation of the last release. After following your instructions I now get "php composer.phar install" And on my testserver I am unable to login to a shell

**Comment by TimZ99** [comment]
I will create a version with composer. wait a moment.

**Comment by TimZ99** [comment]
<REMOVED FILE>

**Comment by SciLor** [comment]
If I am creating a new server, no server entry is created in the "psm_servers" sql table. "Server added" popup comes up, but no new entry is visible.

**Comment by TimZ99** [comment]
hmm, that's weird. Could you enable the debug mode?
add `define('PSM_DEBUG', 'true');` to config.php

**Comment by SciLor** [comment]
<b>Warning</b>:  SQL error: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'post_field' cannot be null in 

Seems to be an easy problem.

PS: I cannot copy paste on the gui of psm.

**Comment by TimZ99** [comment]
I noticed that too but forgot to look into it. Could you make an issue for it?

**Comment by SciLor** [comment]
done

**Comment by TimZ99** [comment]
I meant for the copying part, but this one is also okay 👍

**Comment by SciLor** [comment]
It seems to work, except that #628 won't work.
